core/domain/certificate_assessment_domain.py

```
@@ -246,3 +257,377 @@ def from_dict(
+        """
+        self.attempt_id = attempt_id
+        self.learner_id = learner_id
+        self.total_score = total_score
+        self.attempt_index = attempt_index
+        self.attempt_data = attempt_data
+        self.version_data = version_data
+        self.started_at = started_at
+        self.finished_at = finished_at
+        self.is_submitted = is_submitted
+
+    def validate(self) -> None:
```

Ruff (0.15.20)

[warning] Too many branches (23 > 12)

(PLR0912)